ext/DataInterpolationsExt.jl

```diff
@@ -0,0 +1,11 @@
+module DataInterpolationsExt
+    using Spectra: Spectra, Spectrum, spectrum
+    using DataInterpolations: AbstractInterpolation
+
+    function Spectra.resample(spec::Spectrum, wave_sampled, interp::AbstractInterpolation)
+        p = sortperm(spec.wave) # x-scale must be monitonically increasing
+        flux_sampled = interp(wave_sampled)
+        return spectrum(wave_sampled, flux_sampled; meta = spec.meta)
+    end
```

Comment:
```suggestion
    function Spectra.resample(spec::Spectrum, wave_sampled, interp::AbstractInterpolation; kws...)
        p = sortperm(wave(spec)) # x-scale must be monotonically increasing
        itp = interp(flux(spec)[p], wave(spec)[p]; kws...)
        flux_sampled = itp.(wave_sampled)
        return spectrum(wave_sampled, flux_sampled; meta = spec.meta)
    end
```
`kws...` are used to specify how the interpolation should be extrapolated (if at all)